spec/controllers/results_controller_spec.rb

```diff
@@ -0,0 +1,83 @@
+require 'rails_helper'
+
+RSpec.describe ResultsController, type: :controller do
+  describe 'Filters' do
+    it { should use_before_action(:authenticate_user!) }
+  end
+
+  describe 'GET #index' do
+    context 'without params' do
+      it 'returns all results of user' do
+        user = create :user
+        other_user = create :user
+        number_of_results = 10
+        attachment = create :attachment, user_id: user.id
+        un_authored_attachment = create :attachment, user_id: other_user.id
+        create_list :result, number_of_results, attachment_id: attachment.id
+
+        sign_in user
+        get :index
+        results = controller.instance_variable_get(:@results)
+
+        expect(response).to have_http_status :success
+        expect(results.size).to eq(number_of_results)
+        expect(results.klass.name).to eq(Result.name)
+        expect(results.pluck('DISTINCT attachment_id')).not_to include(un_authored_attachment.id)
+      end
+    end
+
+    context 'when filtered by keyword' do
+      it 'returns all filtered results of user' do
+        user = create :user
+        other_user = create :user
+        keyword = 'e'
+        attachment = create :attachment, user_id: user.id
+        nimble_result = create :result, keyword: 'Nimble', attachment_id: attachment.id
+        company_result = create :result, keyword: 'company', attachment_id: attachment.id
+        vietnam_result = create :result, keyword: 'VietNam', attachment_id: attachment.id
+        un_authored_attachment = create :attachment, user_id: other_user.id
```

Comment:
I'm thinking of renaming this variable to 

```suggestion
        another_user_attachment = create :attachment, user_id: other_user.id
```

But not sure, do you think it is clearer? 🤔 💭

Reply:
Resolved it in e8767c3 ✅